packages/truffle-reporters/compile/index.js

```diff
@@ -0,0 +1,63 @@
+const colors = require("colors");
+const OS = require("os");
+
+module.exports = {
+  compiledSources(options, sources) {
+    if (!sources) return;
+    const logger = options.logger || console;
+    sources.forEach(source => logger.log("    > " + source));
+  },
+
+  finishJob(options, config) {
+    const { globalConfig } = options;
+    const { compilersInfo } = config;
+    const logger = options.logger || console;
+    if (Object.keys(compilersInfo).length > 0) {
+      logger.log(
+        `${OS.EOL}    > artifacts written to ${
+          globalConfig.contracts_build_directory
```

Comment:
Looks like for `truffle compile`, `globalConfig` is passed w/o a `contracts_build_directory` (might be happening in `truffle-workflow-compile/index.js`?)

<img width="364" alt="screen shot 2019-02-26 at 2 40 24 pm" src="https://user-images.githubusercontent.com/7537712/53451927-d7548000-39d4-11e9-893f-130733571f79.png">

Reply:
relevant line in screenshot: `> artifacts written to undefined`